src/state/reducers/dataset/DatasetReducer.js

```diff
@@ -23,6 +23,17 @@ export const datasetListReducer = createReducer(datasetListInitialState, (builde
     .addCase(DATASET_ACTIONS_KEY.ADD_DATASET, (state, action) => {
       delete action.type;
       state.data.push(action);
+    })
+    .addCase(DATASET_ACTIONS_KEY.UPDATE_DATASET_LIST, (state, action) => {
+      state.data = state.data?.map((dataset) => {
+        if (dataset.id === action.datasetId) {
+          return {
+            ...dataset,
+            ...action.datasetData,
+          };
+        }
+        return dataset;
```

Comment:
the syntax can be slightly shorter with ternary operator :slightly_smiling_face: 
```
        dataset.id === action.datasetId
          ? {
              ...dataset,
              ...action.datasetData,
            }
          : dataset
```

Reply:
Anyway, the entire syntax of this case will be rewrited.